coalib/misc/CachingUtilities.py

```diff
@@ -0,0 +1,176 @@
+    """
+    try:
+        os.makedirs(Constants.USER_DATA_DIR, exist_ok=True)
+        return os.path.join(Constants.USER_DATA_DIR, hash_id(identifier))
+    except PermissionError:
+        log_printer.err("Unable to create user data directory '{}'. Continuing"
+                        " without caching.".format(Constants.USER_DATA_DIR))
+    return None
+
+
+def delete_cache_files(log_printer, identifiers):
+    """
+    Delete the cache files after displaying a warning saying the cache
+    is corrupted and will be removed.
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param identifiers: The list of files to be deleted.
+    :return:            True if all the given files were successfully deleted.
+                        False otherwise.
+    """
+    error_files = []
+    for identifier in identifiers:
+        file_path = get_data_path(log_printer, identifier)
+        try:
+            os.remove(file_path)
+        except (OSError, TypeError) as e:
+            error_files.append(hash_id(identifier))
+
+    if len(error_files) > 0:
+        error_files = ", ".join(error_files)
+        log_printer.warn("There was a problem deleting the following "
+                         "files: {}. Please delete them manually from "
+                         "'{}'.".format(error_files, Constants.USER_DATA_DIR))
+        return False
+
+    return True
+
+
+def pickle_load(log_printer, identifier, fallback=None):
+    """
+    Get the data stored in ``filename`` present in the user
+    config directory. Example usage:
+
+    >>> from pyprint.NullPrinter import NullPrinter
+    >>> from coalib.output.printers.LogPrinter import LogPrinter
+    >>> log_printer = LogPrinter(NullPrinter())
+    >>> test_data = {"answer": 42}
+    >>> pickle_dump(log_printer, "test_project", test_data)
+    True
+    >>> pickle_load(log_printer, "test_project")
+    {'answer': 42}
+    >>> pickle_load(log_printer, "nonexistant_project")
+    >>> pickle_load(log_printer, "nonexistant_project", fallback=42)
+    42
+
+
+    :param log_printer: A LogPrinter object to use for logging.
+    :param identifier:  The name of the file present in the user config
+                        directory.
+    :param fallback:    Return value to fallback to in case the file doesn't
+                        exist.
+    :return:            Data that is present in the file, if the file exists.
+                        Otherwise the ``default`` value is returned.
+    """
+    file_path = get_data_path(log_printer, identifier)
+    if file_path == None or not os.path.isfile(file_path):
+        return fallback
+    with open(file_path, "rb") as f:
+        try:
+            return pickle.load(f)
+        except (pickle.UnpicklingError, EOFError) as e:
+            log_printer.warn("The caching database is corrupted and will "
+                             "be removed. Each project will be re-cached "
+                             "automatically in the next run time.")
```

Comment:
this function is not cache specific but the message is, generify the message please :)